src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/ProjectSystem/FallbackProjectManagerTest.cs

```diff
@@ -28,33 +32,44 @@ public FallbackProjectManagerTest(ITestOutputHelper testOutputHelper)
         var languageServerFeatureOptions = TestLanguageServerFeatureOptions.Instance;
         _projectConfigurationFilePathStore = new TestProjectConfigurationFilePathStore();
 
-        var dispatcher = Mock.Of<ProjectSnapshotManagerDispatcher>(MockBehavior.Strict);
-        _projectSnapshotManager = new TestProjectSnapshotManager(ProjectEngineFactoryProvider, dispatcher);
+        _projectManager = new TestProjectSnapshotManager(ProjectEngineFactoryProvider, Dispatcher);
 
-        var projectSnapshotManagerAccessor = Mock.Of<IProjectSnapshotManagerAccessor>(a => a.Instance == _projectSnapshotManager, MockBehavior.Strict);
+        var projectManagerAccessor = StrictMock.Of<IProjectSnapshotManagerAccessor>(a =>
+            a.Instance == _projectManager);
 
-        _fallbackProjectManger = new FallbackProjectManager(_projectConfigurationFilePathStore, languageServerFeatureOptions, projectSnapshotManagerAccessor, WorkspaceProvider, NoOpTelemetryReporter.Instance);
+        _fallbackProjectManger = new FallbackProjectManager(
+            _projectConfigurationFilePathStore,
+            languageServerFeatureOptions,
+            projectManagerAccessor,
+            Dispatcher,
+            WorkspaceProvider,
+            NoOpTelemetryReporter.Instance);
     }
 
-    [Fact]
-    public void DynamicFileAdded_KnownProject_DoesNothing()
+    [UIFact]
```

Comment:
❓why did this have to change to `UIFact`? (I have no idea what UIFact does tbh. I assume it spins up a main thread)

Reply:
UIFact ensures there's a SynchronizationContext, allowing the JTF in our ToolingTestBase to work properly.